### ISSUE TITLE
Consider timingInfo handling in static routing API

### ISSUE THREAD

**Issue by sisidovski** [WICG/service-worker-static-routing-api#19]
In https://www.w3.org/TR/service-workers/#handle-fetch, [timingInfo](https://www.w3.org/TR/service-workers/#ref-for-service-worker-timing-info) is set with start time, fetch event dispatch time appropriately. As the static routing API change the flow of the handle fetch algorithm, we have to define when/what value should be set to timingInfo.

**Comment by yoshisatoyanagisawa** [comment]
In current Chromium implementation, if "cache" source is used, the timing is set like:
https://source.chromium.org/chromium/chromium/src/+/main:content/browser/service_worker/service_worker_cache_storage_matcher.cc;l=118;drc=c686e8f4fd379312469fe018f5c390e9c8f20d0d
I just put random timing info to adjust to the [DidDispatchFetchEvent](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/service_worker/service_worker_main_resource_loader.cc;l=721;drc=705ff26ec9a4b46193ba161c3e90d597acddefdd) interface.

However, we should revisit what information is set there and given as the timing Info.  There might be three options:
Option A: giving new meanings to existing timing Info fields.
Option B: new timing Info field is defined for the static routing API.
Option C: no timing Info are provided for the static routing API.

My recommendation is Option B.
No implementation is needed for Option C, but developers lose the timing Information when the API is used, which sounds not good when the API is widely used.
Option A can be easy to implement, but developers may want to know more details on the route.  Option A cannot fulfill that.
Option B may bring dedicated information for the static routing API, and fix concerns for Option A and C.  Drawback should be design and implementation cost.

**Comment by domenic** [comment]
It looks like this impacts the two properties `workerStart` and `fetchStart` of the navigation timing entries.

Option B involves adding new properties, right? Do you have a proposal for those?

But even if you do option B, we still need to decide what values the existing properties get. Have you thought about what is preferred there? Answering this seems more important, as you can always add new properties later, after gathering more input from partners and the standards community.

To me, setting them both to 0 seems reasonable. (Or whatever default value falls out of the processing, if it is not 0.)

**Comment by yoshisatoyanagisawa** [comment]
Yes, I have drafted [the proposal](https://docs.google.com/document/d/1jWh3LGVc2tZfGS455xSh-0s1LHZxcXO-r0Cznf0oxfg/edit?resourcekey=0-0P4Hvo40gDxe4DSUEgAULg) for Option B.  I suggested to add 4 fields there.  Since timing is got inside the [Handle Fetch](https://w3c.github.io/ServiceWorker/#on-fetch-request-algorithm) algorithm, I suggested to refer the [Service Worker timing](https://w3c.github.io/ServiceWorker/#service-worker-timing) in the [Resource timing](https://w3c.github.io/resource-timing/).

Existing properties are left as-is.  Values may set upon the source.  For the cache source, since we do not call the [Run Service Worker](https://w3c.github.io/ServiceWorker/#run-service-worker-algorithm) algorithm, workerStart is left 0.  The new cache lookup related field will be filled.

As you know, I am working on drafting the Service Worker static routing specification ([PR](https://github.com/w3c/ServiceWorker/pull/1701) that gathers all changes, [working branch](https://github.com/yoshisatoyanagisawa/ServiceWorker/tree/static_routing_api)).  The proposal will be adjusted upon its progress.

**Comment by sisidovski** [comment]
> To me, setting them both to 0 seems reasonable. (Or whatever default value falls out of the processing, if it is not 0.)

Agree. If the router source is `network` or `cache`, setting 0 is reasonable. If the source is `race-network-and-fetch-handler`, wondering what values are appropriate, especially when network wins the race. Should we set the actual values to `workerStart` and `fetchStart`, or we can just set 0 in that case?

**Comment by yoshisatoyanagisawa** [comment]
I suggest to continue the discussion in https://github.com/w3c/resource-timing/issues/389 instead.
